packages/odyssey-react/src/components/Text/index.tsx

```diff
@@ -0,0 +1,167 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import type {
+  FunctionComponent,
+  ReactText,
+  ReactNode,
+  HTMLAttributes,
+} from "react";
+import { useCx, useOmit } from "../../utils";
+import styles from "./Text.module.scss";
+import ScreenReaderText from "../ScreenReaderText";
+
+export interface Props extends HTMLAttributes<HTMLElement> {
```

Comment:
these props feel overly generic? I've also read that extending from `HTMLAttributes<HTMLElement>` can be problematic for additional props we specify that might be part of the native set as well (they won't work). My gut says to use a discriminated union of props that are specific to each `as` value you're allowing below but there could be other clever solutions here might want to do some additional research to see whats out there

Reply:
This is great feedback, will fix!